generic/sosreport.py

```diff
@@ -0,0 +1,145 @@
+                self.skip("Fail to install sosreport required for this test.")
+        if dist.name == 'redhat':
+            if not sm.check_installed("sos") and not sm.install("sos"):
+                self.skip("Fail to install sos package required for this test.")
+        if distro.detect().name == 'SuSE':
+            self.skip('sosreport is not supported on SLES')
+
+    def test(self):
+        self.log.info(
+            "===============Executing sosreport tool test===============")
+        self.run_cmd("sosreport -h")
+        self.run_cmd("sosreport -l")
+        self.run_cmd("sosreport --batch --case-id=test123 -n memory,samba")
+        if process.system_output("sosreport --batch --case-id=test123 -n libraries | "
+                                 "grep libraries", shell=True, ignore_status=True):
+            self.is_fail += 1
+            self.log.info("--skip-plugins option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 -e ntp,numa,snmp")
+        if 'sendmail' not in process.system_output("sosreport --batch --case-id=test123 -e sendmail | "
+                                                   "grep sendmail", shell=True, ignore_status=True):
+            self.is_fail += 1
+            self.log.info("--enable-plugins option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 -o pci,powerpc,procenv,process,processor,kdump")
+        if process.system_output("sosreport --batch --case-id=test123 -o cups | "
+                                 "grep kernel", shell=True, ignore_status=True):
+            self.is_fail += 1
+            self.log.info("--only-plugins option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 -o dlm -k dlm.lockdump")
+        self.run_cmd("sosreport --batch --case-id=test123 -k filesys.dumpe2fs=on")
+        self.run_cmd("sosreport --batch --case-id=test123 --log-size=2000")
+        self.run_cmd("sosreport --batch --case-id=test123 -a")
+        self.run_cmd("sosreport --batch --case-id=test123 --all-logs")
+        self.run_cmd("sosreport --batch --case-id=test123 --build")
+        dir_name = process.system_output("sosreport --batch --case-id=test123 --build | "
+                                         "grep located | cut -d':' -f2", shell=True, ignore_status=True).strip()
+        if not os.path.isdir(dir_name):
+            self.is_fail += 1
+            self.log.info("--build option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 -v")
+        self.run_cmd("sosreport --batch --case-id=test123 --verify")
+        if process.system_output("sosreport --batch --case-id=test123 --quiet -e ntp,numa1", shell=True, ignore_status=True):
+            self.is_fail += 1
+            self.log.info("--quiet option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 --debug")
+        if 'test123' not in process.system_output("sosreport --batch --ticket-number=test123  | "
```

Comment:
Similar to what `run_cmd()` does for `process.system()`, you can have a function for `process.system_output()` as well, in all places below